src/core/GrooveFactory.cpp

```diff
@@ -0,0 +1,31 @@
+
+
+#include "GrooveFactory.h"
+
+#include "Groove.h"
+#include "MidiSwing.h"
+#include "HydrogenSwing.h"
+
+GrooveFactory::GrooveFactory()
+{
+}
+
+/**
+ * Factory method to create grooves classes of the correct type.
+ * grooveType should match the string returned by the grooves nodeName() method
+ *
+ * TODO this is a bit Java-like how does C++ do this kind of thing normally
+ */
+Groove * GrooveFactory::create(QString _grooveType) {
+	if (_grooveType == NULL || _grooveType == "none") {
```

Comment:
In c++, you could do it this way:
```cpp
switch(_grooveType) {
   case "hydrogen":
      return new HydrogenSwing();
   case "midi":
      return new MidiSwing();
  
   default:
      return new Groove();
}
```

Reply:
Talking about this: 

```c++
        if (_grooveType == NULL || _grooveType == "none") {
	     return new Groove();
        }
 	if (_grooveType == "hydrogen") {
 		return new HydrogenSwing();
        }
	if (_grooveType == "midi") {
		return new MidiSwing();
 	}
 	return new Groove();
```

Reply:
No, you can't do that in this way. `QString` can't be used for switch statement directly.

Reply:
^-- What he said.  A candidate for an enumerated type perhaps, but QString can't switch (I looked into this exact scenario for several other use-cases).   More info here: https://stackoverflow.com/a/25873662/3196753